### ISSUE TITLE
Error using Chrome browser-control on MacOS [I fixed it, consider updating README]

### ISSUE THREAD

**Issue by reid-sutherland** [nneonneo/2048-ai#62]
Note: I am on macOS Catalina, although I believe this issue is more chrome-related than mac-related.

Using the instructions in the README, I used this command to open the chrome window:

`google-chrome --remote-debugging-port=9222` 
(where `google-chrome` is an alias for a chrome executable on my machine)

Then ran the script:

`python 2048.py -b chrome`

Resulting in the following error [don't read]:

```
Traceback (most recent call last):
  File "/Users/Reid/opt/anaconda3/lib/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/Users/Reid/opt/anaconda3/lib/python3.7/http/client.py", line 1244, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/http/client.py", line 1290, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/http/client.py", line 1239, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/http/client.py", line 966, in send
    self.connect()
  File "/Users/Reid/opt/anaconda3/lib/python3.7/http/client.py", line 938, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/socket.py", line 727, in create_connection
    raise err
  File "/Users/Reid/opt/anaconda3/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "2048.py", line 154, in <module>
    exit(main(sys.argv[1:]))
  File "2048.py", line 135, in main
    ctrl = ChromeDebuggerControl(args.port)
  File "/Users/Reid/Downloads/2048-ai-master/chromectrl.py", line 30, in __init__
    pages = json.loads(urlopen('http://localhost:%d/json/list' % port).read())
  File "/Users/Reid/opt/anaconda3/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/urllib/request.py", line 1345, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/Users/Reid/opt/anaconda3/lib/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 61] Connection refused>
```

Which I eventually found the fix for. **The command needed to open a chrome window which can be controlled through your script is:**

`google-chrome --remote-debugging-port=9222 --user-data-dir=remote-profile`

Running the window on a separate profile is crucial to allow for debugging, according to [this blog](https://blog.chromium.org/2011/05/remote-debugging-with-chrome-developer.html).

I only posted this because I spent like 45 minutes trying to debug the issue before coming across the simple solution lol. I would consider updating the README and possibly the comments in chromectrl.py

Thanks for the awesome AI, this is incredibly cool. I may use some of your code to develop a Machine Learning variant of the game for fun, of course I will credit your page if I do.

**Comment by nneonneo** [comment]
Thanks. Actually, the problem is likely that you already have Google Chrome running, in which case running `google-chrome` will just switch over to the existing instance instead of starting a new one. Using `--user-data-dir` would create a new profile called `remote-profile`, I think.

I can add some language to the readme to clarify this point.

**Comment by nneonneo** [comment]
Closed in 1e8a12c.

**Comment by reid-sutherland** [comment]
I was certain I had already tried quitting first but I guess I had not! That would explain the issue I had with firefox as well. Thanks!
